File: 250529/Run Length 인코딩/run-length-encoding.py
A = input()
prev_char=A[0]
count=1
result=''
for i in range(1,len(A)):
    if A[i] == prev_char:
        count+=1
    else:
        result+=f"{prev_char}{count}"
        prev_char=A[i]
        count=1
result+=f"{prev_char}{count}"
print(len(result))
print(result)

File: 250529/Run Length 인코딩/run-length-encoding.cpp
#include <iostream>
#include <string>

using namespace std;



int main() {
    string A;
    cin >> A;
    string result="";
    char prev = A[0];
    int count=1;
    
    
    for(int i=1; i<A.length(); i++){
        if(A[i]==prev){
            count++;

        }
        else{
            result+=prev;
            result+=to_string(count);
            prev=A[i];
            count=1;
        }
    }
    result+=prev;
    result+=to_string(count);
    cout << result.length() << endl << result;

    // Please write your code here.

    return 0;
}
